pymeasure/instruments/rohdeschwarz/fsl.py

```diff
@@ -0,0 +1,259 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+import numpy as np
+from pymeasure.instruments.validators import strict_discrete_set
+from pymeasure.instruments import Instrument
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class FSL(Instrument):
+    """
+    Represents a Rohde&Schwarz FSL spectrum analyzer.
+
+    All physical values that can be set can either be as a string of a value
+    and a unit (e.g. "1.2 GHz") or as a float value in the base units (Hz,
+    dBm, etc.).
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super(FSL, self).__init__(
+            resourceName, "Rohde&Schwarz FSL", includeSCPI=True, **kwargs
+        )
+
+    # Frequency settings ------------------------------------------------------
+
+    freq_span = Instrument.control(
+        "FREQ:SPAN?",
+        "FREQ:SPAN %s",
+        "Frequency span in Hz.",
+    )
+
+    freq_center = Instrument.control(
+        "FREQ:CENT?",
+        "FREQ:CENT %s",
+        "Center frequency in Hz.",
+    )
+
+    freq_start = Instrument.control(
+        "FREQ:STAR?",
+        "FREQ:STAR %s",
+        "Start frequency in Hz.",
+    )
+
+    freq_stop = Instrument.control(
+        "FREQ:STOP?",
+        "FREQ:STOP %s",
+        "Stop frequency in Hz.",
+    )
+
+    attenuation = Instrument.control(
+        "INP:ATT?",
+        "INP:ATT %s",
+        "Attenuation in dB.",
+    )
+
+    @property
+    def rbw(self):
+        """Resolution bandwidth in Hz. Can be set to 'AUTO'."""
+        return self.values("BAND:RES?")[0]
+
+    @rbw.setter
+    def rbw(self, value):
+        if type(value) is str and value.upper() == "AUTO":
+            self.write("BAND:RES:AUTO ON")
+        else:
+            self.write(f"BAND:RES {value}")
+
+    @property
+    def vbw(self):
+        """Video bandwidth in Hz. Can be set to 'AUTO'."""
+        return self.values("BAND:VID?")[0]
+
+    @vbw.setter
+    def vbw(self, value):
+        if type(value) is str and value.upper() == "AUTO":
+            self.write("BAND:VID:AUTO ON")
+        else:
+            self.write(f"BAND:VID {value}")
+
+    # Sweeping ----------------------------------------------------------------
+
+    @property
+    def sweep_time(self):
+        """Sweep time in s. Can be set to 'AUTO'."""
+        return self.values("SWE:TIME?")[0]
+
+    @sweep_time.setter
+    def sweep_time(self, value):
+        if type(value) is str and value.upper() == "AUTO":
+            self.write("SWE:TIME:AUTO ON")
+        else:
+            self.write(f"SWE:TIME {value}")
+
+    continuous_sweep = Instrument.control(
+        "INIT:CONT?",
+        "INIT:CONT %s",
+        "Continuous (True) or single sweep (False)",
+        validator=strict_discrete_set,
+        values=[True, False],
+        set_process=lambda x: "ON" if x else "OFF",
+        get_process=lambda x: bool(x),
```

Comment:
not sure what you're aiming for here and what the device returns on command_get, but maybe
```
        strict_discrete_set,
        values={True: "ON", False: "OFF"},
        map_values=True,
```

Reply:
@bleykauf same here - ping so you see this.

Reply:
Implemented in 97198bc.